api/routes/bond.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class App < Roda
+  route('bond') do |r|
+    r.on String do |url_stub_id|
+      bond = Bond.where(url_stub_id: url_stub_id).first
+
+      if bond.nil?
+        response.status = 404
+        response.write("<h1>Sorry, I don't know what you mean </h1>")
+        r.halt
+      end
```

Comment:
so let me get this straight:

the stuff inside `r.on` get executed before hand no matter if it is a `get` or `post` ? Starting to like roda a bit ... :+1: